components/steering_wheel/HW/include/HW_gpio.h

```diff
@@ -16,12 +16,32 @@ extern "C" {
 #include "SystemConfig.h"
 
 
+/******************************************************************************
+ *                             T Y P E D E F S
+ ******************************************************************************/
+
+typedef enum

```

Comment:
we should eliminate these dependencies on the HAL. No need to use `GPIO_PIN_RESET` and `SET`, just use `0U` and `1U`. Don't just typedef HW_GPIO_TypeDef, redefine the type here so that we have a reference.